test/test_vmap.py

```diff
@@ -433,5 +445,62 @@ def foo(x):
         vmap(foo, in_dims=(0,))(torch.randn(2, 3))
         vmap(foo, in_dims=(1,))(torch.randn(2, 3))
 
+    def test_fallback_sub(self):
+        # NB: One day we will implement a batching rule for torch.sub.
+        # If/when we do, this test should be replaced to test the fallback
+        # path on another operator to avoid bitrot.
+        x = torch.randn(5, 7, 11)
+        y = torch.randn(5, 7, 11)
+
+        # Test the fallback path raises a warning
+        with warnings.catch_warnings(record=True) as wa:
+            result = vmap(torch.sub)(x, y)
+            self.assertEqual(len(wa), 2)
+            self.assertRegex(str(wa[-1].message),
+                             r'falling back to slow \(for loop and stack\) implementation')
+            self.assertEqual(result, x - y)
+
+        # fallback on torch.sub
+        x = torch.randn(7, 11, 5)
+        y = torch.randn(5, 7, 11)
+        result = vmap(torch.sub, (2, 0))(x, y)
+        self.assertEqual(result, x.permute(2, 0, 1) - y)
+
+        # fallback on torch.sub, nested vmap
+        x = torch.randn(7, 11, 5)
+        y = torch.randn(5, 7, 11)
+        result = vmap(vmap(torch.sub), (2, 0))(x, y)
+        self.assertEqual(result, x.permute(2, 0, 1) - y)
+
+        # big batch size (total 10000)
+        x = torch.randn(100, 10, 10, 5)
+        y = torch.randn(100, 10, 10)
+        result = vmap(vmap(vmap(torch.sub)))(x, y)
+        self.assertEqual(result, x - y.view(100, 10, 10, 1))
+
+    def test_fallback_masked_fill(self):
+        # NB: One day we will implement a batching rule for index_add
+        # If/when we do, this test should be replaced to test the fallback
+        # path on another operator to avoid bitrot.
```

Comment:
It would be better if you could detect if this situation occurred :) (Maybe you could use one of the apis used in `test_dispatch.py`, or a new API, to test that `index_add` doesn't have an explicit VMap key installed. Not sure it's worth the faffing about.)

Reply:
We can detect it actually -- the fallback emits a warning message and we test for the warning message in `test_fallback_masked_fill`. If someone adds a batching rule for `masked_fill`, then the test will fail and the developer will read this comment :)